packages/core-api/src/routing/hooks.tsx

```diff
@@ -41,14 +58,20 @@ export function useRouteRef<Params extends AnyParams>(
     | ExternalRouteRef<Params, any>,
 ): RouteFunc<Params> | undefined {
   const sourceLocation = useLocation();
-  const resolver = useContext(RoutingContext);
+  const versionedContext = useContext(
+    getGlobalSingleton<Context<RoutingContextType>>('routing-context'),
+  );
+  const resolver = versionedContext?.atVersion(1);
   const routeFunc = useMemo(
     () => resolver && resolver.resolve(routeRef, sourceLocation),
     [resolver, routeRef, sourceLocation],
   );
 
-  if (!routeFunc && !resolver) {
-    throw new Error('No route resolver found in context');
+  if (!versionedContext) {
+    throw new Error('useRouteRef used outside of routing context');
+  }
+  if (!resolver) {
+    throw new Error('RoutingContext v1 not available');
```

Comment:
could have a long and helpful error message saying what the user can do to fix it

Reply:
Kind of need to get to the point where we actually have some instructions to give here tbh. This piece of code is supposed to end up in the consumer library, so it's somewhat a thing that "shouldn't happen". In reality I think we want to have a URL to link to that explains the relationship between the versions of the different core libraries and that perhaps your plugin is too far behind the app.

For now I think it's fine to leave as is, as we still have the verbose `useApi` error message in place, which is the place you pretty much always end up hitting the issue first.